src/nvim/terminal.c

```diff
@@ -161,6 +161,8 @@ static Map(int, int) *color_indexes;
 static int default_vt_fg, default_vt_bg;
 static VTermColor default_vt_bg_rgb;
 
+static void buf_set_term_title(buf_T *buf, String title);
```

Comment:
Though it only says what you should do, not what gendeclarations.lua does, how and where it is called, etc.